cmd/buildkitd/config/config.go

```diff
@@ -112,6 +112,7 @@ type ContainerdConfig struct {
 	Labels    map[string]string `toml:"labels"`
 	Platforms []string          `toml:"platforms"`
 	Namespace string            `toml:"namespace"`
+	Runtime   string            `toml:"runtime"`
```

Comment:
Shouldn't this support options for runc binary path, cgroup driver, etc?
I also guess this should be `map[string]interface{}` to support multiple runtime classes in the future.

i.e., The config structure should be similar to the config structure of containerd/CRI.

Reply:
I am not sure how to pass that to containerd. What I just wanted to achieve is to be able to override `client.defaultRuntime`. Because there are *cough, cough* platforms where runtime isn't mature enough to be declared _The Default_.

The lack of such configuration option leaded to FreeBSD runtime being hardcoded in buildkit. I'm in the same boat with macOS.

Reply:
So, my idea is: with these changes, one can configure runtime options in containerd config, while using buildkit to select needed runtime. Or, option added here can just be used to specify path path to runtime binary. Exposing the full runtime configuration through buildkit config is too much extending the scope of the task.

Reply:
I think I missed this originally. We should probably make sure to cover this is in a first pass - I wrote up a possible implementation: https://github.com/jedevc/buildkit/commit/22d9bb104de0261357194179d7ac17298f817e90. @AkihiroSuda @tonistiigi wdyt? (I took inspiration from how CRI does it today). @slonopotamus you happy to pull that change into your branch (or I can force-push)?

Without a way to configure the runtime, I think the feature is quite limited (e.g. I have a use case for wanting to override the runc binary used by the `io.containerd.runc.v2` runtime, which is possible by using the `binary_name` option). Containerd doesn't seem to have any other way to configure this, so we need to allow this configuration in buildkit if we want to allow users this kind of freedom.

Even if we don't take this now, we need to provide a place in config fields to allow this kind of configuration. Previously we had:

```toml
[worker.containerd]
runtime = "io.containerd.runc.v2"
```

I think something like this should work easily:

```toml
[worker.containerd.runtime]
name = "io.containerd.runc.v2"
options = { binary_name = "foobar" }
```

Reply:
@jedevc I'm perfectly fine if you create a separate PR. I just need _a_ way to specify which runtime to use, it isn't really important what lines one would need to write into config file, it is easy to adapt.

Reply:
Opened in https://github.com/moby/buildkit/pull/4279.